atlasdb-autobatch/src/main/java/com/palantir/atlasdb/autobatch/BatchSizeLogger.java

```diff
@@ -56,7 +56,7 @@ public void markBatchProcessed(long batchSize) {
         total += batchSize;
         counter++;
         if (shouldFlush.getAsBoolean()) {
-            log.info("The autobatcher with identifier {} has just processed a batch of size {}."
```

Comment:
this is the one I'm only 60% happy about; It's definitely _very_ spammy, but does find people making stupid "hey can I just get the world in this one transaction" queries.

The rest of this I 100% agree on.

Reply:
I must be reading this wrong, but this seems to log once every 10 seconds no matter what. Wouldn't it be better to just expose `getCurrentAverage()` as a gauge?

Reply:
Yep, I'd say this line has been useful in figuring out problems in the past. I think it was initially implemented this way because the Gradle project for `atlasdb-autobatch` didn't use to have `MetricRegistry` on the class-path - I think for OSS debugging at least we've mainly looked at the average in a gauge-like way.

@clockfort do individual batch sizes matter / have they mattered on any of the cases you've debugged with this?

Reply:
I made it a metric-registry. I didn't want to bring in some of the constants like AtlasDbNames since that would put a dependency on atlasdb-client. 

Is a Meter good enough for this? That gives you average batch sizes over time, plus histograms.

Reply:
I made it a histogram instead, since a meter wasn't quite right for the measurement tracking. Now it aligns nicely with the original logs.

Reply:
Yeah, a histogram would work well!